### ISSUE TITLE
CPU profiles: add the `.comment` value as rizin comment near the flag, if defined

### ISSUE THREAD

**Issue by ret2libc** [rizinorg/rizin#1671]
The .comment value in the CPU platform profile seems to be completely ignored.
I think instead it should be added as a Rizin comment to the address where the port is used.

Moreover, a `.comment` should be available in architecture profiles as well, because those may be useful as well.

**Comment by officialcjunior** [comment]
>I think instead it should be added as a Rizin comment to the address where the port is used.

Alright, so I think adding comments from Platform profiles [here at canalysis](https://github.com/rizinorg/rizin/blob/dev/librz/core/canalysis.c#L6087) at the place where the port/flag was defined was the wrong way to do it, since it doesn't show up at the place it is used. I think we'll remove this part?

As a fix, I believe that the comments can be unpacked and added over here at [disasm.c](https://github.com/rizinorg/rizin/blob/dev/librz/core/disasm.c#L3425) and that would make sense.

**Comment by ret2libc** [comment]
I see... so we are using the comment at the place where the port/mmio is defined, not where it is used. I'm not sure, but I think seeing the description where the port/mmio is used is better than having it where the port/mmio is defined.

@thestr4ng3r @XVilka @wargio WDYT?

**Comment by XVilka** [comment]
@ret2libc I think it's good to have the same comment in both places actually - in place of definition (actual location) and the place of use.

**Comment by wargio** [comment]
i agree. probably in both places is better.

**Comment by officialcjunior** [comment]
So, I've made a short API to return the comment and was looking for the best place to call it and add the comment.

A place like `ds_print_sysregs` sounds alright, but is the places where the ports are used really confined to just one opcode type?

**Comment by wargio** [comment]
> So, I've made a short API to return the comment and was looking for the best place to call it and add the comment.
> 
> A place like `ds_print_sysregs` sounds alright, but is the places where the ports are used really confined to just one opcode type?

Yes, it can be either load/store/IO opcodes

**Comment by officialcjunior** [comment]
> > So, I've made a short API to return the comment and was looking for the best place to call it and add the comment.
> > A place like `ds_print_sysregs` sounds alright, but is the places where the ports are used really confined to just one opcode type?
> 
> Yes, it can be either load/store/IO opcodes

I think we'll get another function `ds_print_arch_ports` or something and add the comment if the `ds->analop.type` is any of those opcode types?

**Comment by ret2libc** [comment]
Any progress on this @officialcjunior ?

**Comment by officialcjunior** [comment]
Hey, hey

> Any progress on this @officialcjunior ?

Yep, I had written the code for that a while ago and was currently trying to figure how to make/find a binary to actually test it.

Basically, I just want some instruction to load a value from one of the `.address`-es here at `asm/platforms` -- but I just can't seem to do that, probably due to my idiocy. First thing I tried to do was to set up some ARM compiler (all the added platform profiles are for ARM) to make one myself, but I couldn't produce an instruction that could `LDR` from a literal yet. (I was trying to tinker and figure it out myself without asking y'all, to be honest)

So, yeah, I just want to test it. Any other good ideas are welcome.

**Comment by ret2libc** [comment]
@officialcjunior are you looking for something like `ldr.w sl, [0x00000434]`? have you looked in `./test/bins/arm/elf/hello_world` ?

**Comment by officialcjunior** [comment]
Yes. That's what I was trying to do at the beginning: I wanted an instruction just like that, taking a value from say `0x7e215048` which is `AUX_MU_IIR_REG` from the BCM2835's platform profile.

I looked up and wrote a simple code which seems to do just that:
```asm
.cpu cortex-m3
.thumb
ldr r1, =0x7e215048
.end
```
Which when assembled gave a disassembly like this:

```asm
            ;-- .text:                                                                                                                 
            0x08000034      ldr   r1, [0x08000038]                     ; [0x8000038:4]=0x7e215048 ; [01] -r-x section size 8 named .te
            0x08000036      movs  r0, r0                                                                                               
            0x08000038      str   r0, [r1, r1]                                                                                         
            0x0800003a      ldrb  r1, [r4, 0x18]
```
So, now I was figuring out how `0x7e215048` is calculated (`[0x8000038:4]=0x7e215048`). Yeah, you read 4 bytes from `0x8000038` to do that, but haven't figured how to write the code yet.  I had made a small API which returned the comment of the port when given the address.

Right now, I'm thinking whether this is really the right way to do this 🤔

**Comment by officialcjunior** [comment]
I've opened a draft PR to speed things up a bit.
So far, I've made an API which returns the .comment when the address of the port is passed to it.

I just would like to know what you think should be done next :)

**Comment by officialcjunior** [comment]
I was checking whether the de-referenced value (i.e 0x7e215048 on the above example code) was stored somewhere.
I couldn't find it out, so I just put that address in a new variable (at here in [disasm.c](https://github.com/rizinorg/rizin/blob/dev/librz/core/disasm.c#L4019)) I made under `ds` and tried to pass it to the new API I made and add the comment.

Which seems to work:

```bash
            ;-- section..text:
┌ sym..text ();
│ bp: 0 (vars 0, args 0)
│ sp: 0 (vars 0, args 0)
│ rg: 0 (vars 0, args 0)
│           0x08000034      ldr   r1, [0x08000038]                     ; [0x8000038:4]=0x7e215048 AUX_MU_IIR_REG ; MMIO Mini UART Interrupt Identify -  ; [01] -r-x section size 12 named .text
│           0x08000036      ldr   r2, [0x0800003c]                     ; [0x800003c:4]=0x5d00014

```

I think this is good enough to be pushed to the PR, it seems?
